File: Server/src/main/java/lk/wnl/wijeya/entity/ProductPaper.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "product_paper")
public class ProductPaper {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "paper_id", nullable = false)
    private Paper paper;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Column(name = "quentity")
    private Integer quantity;

    @Column(name = "linecost", precision = 7, scale = 2)
    private BigDecimal linecost;

}

File: Client/src/app/entity/ProductPaper.ts
import {Paper} from "./Paper";

export class ProductPaper {
  paper?: Paper;
  lineCost?: number;
  quantity?: number;
  id?: number;
  name?: string;

  constructor(id: number, paper: Paper,  quantity: number,lineCost: number) {
    this.id = id;
    this.paper = paper;
    this.lineCost = lineCost;
    this.quantity = quantity;
  }


}
